<?xml version="1.0"?>
<launch>

  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <group ns="ugr/car">


    <!-- Load the hardware interface -->
    <node name="orion_hardware_interface" pkg="ugr_ros_control" type="orion_hw_main"
      output="screen" launch-prefix="$(arg launch_prefix)">

      <remap from="/output/can" to="/ugr/can/mc/tx" />

      <!-- Encoder 0: right -->
      <!-- Encoder 1: left -->
      <remap from="/output/vel0" to="/ugr/car/sensors/hwinterface/encoder0" />
      <remap from="/output/vel1" to="/ugr/car/sensors/hwinterface/encoder1" />

      <param name="wheel_diameter" value="0.4064" />
      <param name="gear_ratio" value="3.405" />

      <!-- Torque vectoring -->
      <param name="use_torque_vectoring" value="false" />
      <param name="max_dT" value="4.0" />
      <param name="l_wheelbase" value="2" />
      <param name="COG" value="0.5" />
      <param name="Cyf" value="444" />
      <param name="Cyr" value="444" />
      <param name="m" value="320" />
      
      <!-- PID Torque vectoring -->
      <param name="Kp" value="1.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />


    </node>

    <!-- Load controller settings -->
    <rosparam file="$(find ugr_ros_control)/orion_control/config/orion_controllers.yaml" command="load" />

    <!-- Load controller manager -->
    <node name="ros_control_controller_manager" pkg="controller_manager" type="spawner"
      respawn="false"
      output="screen"
      args="--stopped joint_state_controller drive_effort_controller drive_velocity_controller steering_position_controller" />
  </group>

</launch>